exercises/affine-cipher/description.md

```diff
@@ -10,61 +10,54 @@ its new numeric value. Although all mono-alphabetic ciphers are weak,
 the affine cipher is much stronger than the atbash cipher,
 because it has many more keys.
 
+## Encryption
+
 The encryption function is:
 
-  `E(x) = (ax + b) mod m`
-  -  where `x` is the letter's index from 0 - length of alphabet - 1
-  -  `m` is the length of the alphabet. For the roman alphabet `m == 26`.
-  -  and `a` and `b` make the key
+  `E(x) = (ai + b) mod m`
+  -  where `i` is the letter's index from `0` to the length of the alphabet - 1
+  -  `m` is the length of the alphabet. For the roman alphabet `m` is `26`.
+  -  `a` and `b` are integers which make the encryption key
+
+Values `a` and `m` must be *coprime* (or, *relatively prime*) for automatic decryption to succeed,
+ie. they have number `1` as their only common factor (more information about coprime numbers
+can be found [here](https://en.wikipedia.org/wiki/Coprime_integers)). In case `a` is not coprime to `m` your
+program should return status `1` and `"Error: a and m must be coprime."`. Otherwise it should encrypt or
+decrypt with the provided key.
+
+For the purpose of this exercise, digits are valid input but they are not encrypted. Spaces and punctuation
+characters are excluded. Ciphertext is written out in groups of fixed length separated by space,
+the traditional group size being `5` letters. This is to make it harder to guess encrypted text based
+on word boundaries.
+
+## Decryption
 
 The decryption function is:
 
-  `D(y) = a^-1(y - b) mod m`
+  `D(y) = (a^-1)(y - b) mod m`
   -  where `y` is the numeric value of an encrypted letter, ie. `y = E(x)`
-  -  it is important to note that `a^-1` is the modular multiplicative inverse
+  -  it is important to note that `a^-1` is the modular multiplicative inverse (MMI)
      of `a mod m`
-  -  the modular multiplicative inverse of `a` only exists if `a` and `m` are
-     coprime.
+  -  the modular multiplicative inverse only exists if `a` and `m` are coprime.
 
-To find the MMI of `a`:
+The MMI of `a` is `x` such that the remainder after dividing `ax` by `m` is `1`:
 
-  `an mod m = 1`
-  -  where `n` is the modular multiplicative inverse of `a mod m`
+  `ax mod m = 1`
 
 More information regarding how to find a Modular Multiplicative Inverse
 and what it means can be found [here.](https://en.wikipedia.org/wiki/Modular_multiplicative_inverse)
```

Comment:
**optional**, since it was already this way before this PR: another use of "here" as link text

Reply:
Updated too.